recipes/ginkgo/all/conanfile.py

```diff
@@ -0,0 +1,110 @@
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class GinkgoConan(ConanFile):
+    name = "ginkgo"
+    license = "BSD-3-Clause"
+    homepage = "https://github.com/ginkgo-project/ginkgo"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "High-performance linear algebra library for manycore systems, with a focus on sparse solution of linear systems."
+    topics = ("hpc", "linear-algebra")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "openmp": [
+        True, False], "cuda": [True, False]}
+    default_options = {"shared": False, "fPIC": False,
+                       "openmp": False, "cuda": False}
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "16",
+            "gcc": "5.4",
+            "clang": "3.9",
+            "apple-clang": "10.0",
+            "intel": "18"
+        }
+
+    def configure(self):
+        if (self.settings.os == "Windows") or self.options.shared:
```

Comment:
There's a fine difference between these two about how Conan calculates package_id (which are used to generate the binaries)

```suggestion
    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def configure(self):
        if self.options.shared:
```